### DISCUSSION TITLE
Read logs from external disk (S3 Bucket Case)

### DISCUSSION THREAD

**Discussion by rodrigolopespt** [opcodesio/log-viewer#230] [Q&A]
Any way of reading log files from an s3 disk instead of from local storage/logs?

**Comment by arukompas** [comment]
@rodrigolopespt hey! It might be very slow & inefficient to index & read logs from a remote source, hence I haven't done this yet. Log Viewer does a lot of `fseek()` and `fgets()` calls to incrementally read the file line by line and to index the datetime, type, and location of each file. It does **not** load the whole file into memory, and that's why Log Viewer can easily parse 1 GB+ log files with just ~40 MB of RAM used. And for that reason, I think it would be too inefficient from a remote location like S3 storage.

That's just my assumption, though, I haven't actually tried it.
